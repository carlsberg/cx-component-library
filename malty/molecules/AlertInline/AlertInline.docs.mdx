---
labels: ['react', 'typescript', 'AlertInline']
description: 'An AlertInline component.'
---
