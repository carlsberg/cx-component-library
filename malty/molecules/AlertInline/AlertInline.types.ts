import { SpaceProps } from '@carlsberggroup/malty.utils.space';
import { ReactElement } from 'react';

export interface AlertInlineProps extends SpaceProps {
  icon?: ReactElement;
  size?: AlertInlineSize;
  color: AlertInlineColor;
  dataQaId?: string;
  firstAction?: () => void;
  firstActionText?: string;
  secondAction?: () => void;
  secondActionText?: string;
  message: string;
  title?: string;
}

export enum AlertInlineColor {
  Notification = 'notification-strong',
  NotificationLight = 'notification-light',
  Alert = 'alert-strong',
  AlertLight = 'alert-light',
  Success = 'success',
  SuccessLight = 'success-light',
  Fail = 'fail',
  FailLight = 'fail-light'
}

export enum AlertInlineSize {
  Compact = 'small',
  Default = 'medium'
}
