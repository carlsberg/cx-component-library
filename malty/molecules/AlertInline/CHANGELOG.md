# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.2.0 (2024-05-31)

### Bug Fixes

- Run auto fixable lint based on new `eslint` configuration ([#501](https://github.com/CarlsbergGBS/cx-component-library/issues/501)) ([9ee94cc](https://github.com/CarlsbergGBS/cx-component-library/commit/9ee94cc2cd407f717e62d8857b5f20a74e7bbec4))

### Features

- add tsconfig files and update package.json files for each component ([3bfc644](https://github.com/CarlsbergGBS/cx-component-library/commit/3bfc644e1cfc9dbb7cf7a0469e25fce055b53240))
- AlertInline and Toast ([#182](https://github.com/CarlsbergGBS/cx-component-library/issues/182)) ([8f344e3](https://github.com/CarlsbergGBS/cx-component-library/commit/8f344e392a6b339d3f159c3e8a2958ef5ddabd71))
- **alertInline:** [DSM-730] improve word breaking ([#755](https://github.com/CarlsbergGBS/cx-component-library/issues/755)) ([a938911](https://github.com/CarlsbergGBS/cx-component-library/commit/a9389112782d5603b457dae46ae798b36f72c91f))
- change organization name ([a0aa428](https://github.com/CarlsbergGBS/cx-component-library/commit/a0aa428f43138af5707a3ff4c0e36b7c056f02e6))
- **DSM-164:** changes sizes name & font size ([#191](https://github.com/CarlsbergGBS/cx-component-library/issues/191)) ([64df0e9](https://github.com/CarlsbergGBS/cx-component-library/commit/64df0e92198d988fd75bd0da01b9c01a90a0c713))
- **DSM-248:** Headline HTML tags ([#184](https://github.com/CarlsbergGBS/cx-component-library/issues/184)) ([305f5e0](https://github.com/CarlsbergGBS/cx-component-library/commit/305f5e0fa92d5813f4877b338ad827a8c347d452))
- **DSM-256:** link ([#279](https://github.com/CarlsbergGBS/cx-component-library/issues/279)) ([7185428](https://github.com/CarlsbergGBS/cx-component-library/commit/718542872ecffe4a580bfeb8bf71712317f8bad4))
- **general:** [DSM-953] refactor icons implementation ([#892](https://github.com/CarlsbergGBS/cx-component-library/issues/892)) ([08d259f](https://github.com/CarlsbergGBS/cx-component-library/commit/08d259f476ce97e9db8ac41d5eb4773eabca0d37))
- **global:** [DSM-784] add space props in all components ([#842](https://github.com/CarlsbergGBS/cx-component-library/issues/842)) ([9f03f59](https://github.com/CarlsbergGBS/cx-component-library/commit/9f03f59e7518a99abcea06bacbee989d1ddb0465))
- **global:** migrate from bit to lerna ([a0110c8](https://github.com/CarlsbergGBS/cx-component-library/commit/a0110c8831370dc762c193b17cc593eed381f990))
- **global:** update storybook files to be compliant with v7 ([#970](https://github.com/CarlsbergGBS/cx-component-library/issues/970)) ([c61d4b4](https://github.com/CarlsbergGBS/cx-component-library/commit/c61d4b40a0755becf942ad3f28758a159f8c54e4))
- New Chip component and minor changes ([#239](https://github.com/CarlsbergGBS/cx-component-library/issues/239)) ([ce92ab0](https://github.com/CarlsbergGBS/cx-component-library/commit/ce92ab0543b96296952a29e2d43d0df29d4e188f))
- new icon colors ([#215](https://github.com/CarlsbergGBS/cx-component-library/issues/215)) ([9247d33](https://github.com/CarlsbergGBS/cx-component-library/commit/9247d3370bcde78432033ba8b23837f97e0b0dc0))
- remove empty composition files ([f8173f4](https://github.com/CarlsbergGBS/cx-component-library/commit/f8173f4a2ecbf80bb7b6ffe848c023ae31819c2d))
- Use new `eslint-config-carlsberg` with streamlined way to create components ([#559](https://github.com/CarlsbergGBS/cx-component-library/issues/559)) ([a459332](https://github.com/CarlsbergGBS/cx-component-library/commit/a45933215bc0b523220743f17d988d7f1ad5fce6))
