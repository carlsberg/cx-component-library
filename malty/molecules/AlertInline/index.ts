export { AlertInline } from './AlertInline';
export { AlertInlineColor, AlertInlineSize } from './AlertInline.types';
export type { AlertInlineProps } from './AlertInline.types';
