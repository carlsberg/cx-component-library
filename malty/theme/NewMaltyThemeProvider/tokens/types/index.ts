export type { PrimitiveBorderRadiusTokens, SemanticBorderRadiusTokens } from './borderRadius.types';
export type { PrimitiveBorderTokens, SemanticBorderTokens } from './borders.types';
export type { PrimitiveColorsTokens, SemanticColorsTokens } from './colors.types';
export type { GridTokens } from './grid.types';
export type { OpacityTokens } from './opacity.types';
export type { ShadowTokens } from './shadows.types';
export type {
  AdditionalPrimitives,
  BorderRadiusTokens,
  ColorTokens,
  SemanticOverrides,
  SemanticTokens
} from './shared.types';
export type { SizeTokens } from './sizes.types';
export type { TypographyTokens } from './typography.types';
