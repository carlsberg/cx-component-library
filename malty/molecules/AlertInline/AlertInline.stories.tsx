import { IconName } from '@carlsberggroup/malty.atoms.icon';
import { action } from '@storybook/addon-actions';
import { Meta, Story } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import { AlertInline as AlertInlineComponent } from './AlertInline';
import { AlertInlineColor, AlertInlineProps, AlertInlineSize } from './AlertInline.types';

export default {
  title: 'Information/Alert Inline ',
  component: AlertInlineComponent,
  parameters: {
    importObject: 'InlineAlert',
    importPath: '@carlsberggroup/malty.molecules.inline-alert',
    variants: ['icon', 'action', 'title', 'complete']
  },
  argTypes: {
    title: {
      control: 'text',
      description: 'Title'
    },
    message: {
      control: 'text',
      description: 'message'
    },

    icon: {
      description: 'When selected, Alert label will contain the selected icon',
      options: Object.values(IconName),
      control: {
        type: 'select'
      }
    },
    size: {
      description: 'Alert Height size - Only for In Line Alert type without actions. Options are',
      options: Object.keys(AlertInlineSize),
      mapping: AlertInlineSize,
      control: {
        type: 'select',
        label: Object.values(AlertInlineSize)
      },
      table: {
        defaultValue: {
          summary: 'AlertInlineSize.Default'
        }
      }
    },
    color: {
      description: 'Alert colors, from design predefined colors, as follows.',
      options: Object.keys(AlertInlineColor),
      mapping: AlertInlineColor,
      control: {
        type: 'select',
        label: Object.values(AlertInlineColor)
      },
      table: {
        defaultValue: {
          summary: 'AlertInlineColor.Notification'
        }
      }
    },
    dataQaId: {
      control: 'text',
      description: 'Alert data-qi-id, can be',
      table: { defaultValue: { summary: 'none' } }
    },
    firstAction: {
      description: 'Function to be executed when first action is clicked'
    },
    secondAction: {
      description: 'Function to be executed when second action is clicked'
    },
    firstActionText: {
      control: 'text',
      description: 'First Action text',
      table: { defaultValue: { summary: 'none' } }
    },
    secondActionText: {
      control: 'text',
      description: 'Second Action text',
      table: { defaultValue: { summary: 'none' } }
    }
  }
} as Meta;
const StyledContainer = styled.div`
  height: 200px;
  width: 100%;
`;
const Template: Story<AlertInlineProps> = ({ ...args }) => (
  <StyledContainer>
    <AlertInlineComponent {...args} />
  </StyledContainer>
);

export const AlertInline = Template.bind({});
const params = new URLSearchParams(window.location.search);
const variant = params.get('variant');
switch (variant) {
  case 'icon':
    AlertInline.args = {
      message: 'Hello, Im the In Line Alert! Play with me.',
      icon: IconName.Information,
      size: AlertInlineSize.Default,
      color: AlertInlineColor.Notification,
      dataQaId: 'inline-alert'
    };
    break;
  case 'title':
    AlertInline.args = {
      title: 'Title Alert',
      message: 'Hello, Im the In Line Alert! Play with me.',
      size: AlertInlineSize.Default,
      color: AlertInlineColor.Notification,
      dataQaId: 'inline-alert'
    };
    break;
  case 'action':
    AlertInline.args = {
      message: 'Hello, Im the In Line Alert! Play with me.',
      size: AlertInlineSize.Default,
      color: AlertInlineColor.Notification,
      dataQaId: 'inline-alert',
      firstAction: action('First Action clicked'),
      firstActionText: 'First Action',
      secondAction: action('Second Action clicked'),
      secondActionText: 'Second Action'
    };
    break;
  case 'complete':
    AlertInline.args = {
      title: 'Title Alert',
      message: 'Hello, Im the In Line Alert! Play with me.',
      size: AlertInlineSize.Default,
      color: AlertInlineColor.Notification,
      dataQaId: 'inline-alert',
      firstAction: action('First Action clicked'),
      firstActionText: 'First Action',
      secondAction: action('Second Action clicked'),
      secondActionText: 'Second Action'
    };
    break;

  default:
    AlertInline.args = {
      message: 'Hello, Im the In Line Alert! Play with me.',
      size: AlertInlineSize.Default,
      color: AlertInlineColor.Notification,
      dataQaId: 'inline-alert'
    };
    break;
}
