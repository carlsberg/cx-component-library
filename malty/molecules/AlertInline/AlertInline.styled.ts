import styled from 'styled-components';
import { AlertInlineColor, AlertInlineSize } from './AlertInline.types';

export const StyledWrapper = styled.div`
  width: 100%;
  position: relative;
  box-sizing: border-box;
`;

export const StyledAlertInLineWrapper = styled(StyledWrapper)`
  margin-top: ${({ theme }) => theme.sizes['2xs'].value}; ;
`;

export const StyledContainer = styled.div<{
  color?: AlertInlineColor;
  size?: AlertInlineSize;
}>`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  box-sizing: border-box;
  row-gap: ${({ theme }) => theme.sizes['4xs'].value};
  word-break: break-all;

  svg {
    flex-shrink: 0;
  }
`;

export const StyledAlertInLine = styled(StyledContainer)<{
  color: AlertInlineColor;
  size?: AlertInlineSize;
  hasTitle?: boolean;
  hasActions?: boolean;
  hasIcon?: boolean;
}>`
  background-color: ${({ color, theme }) => theme.colors.colours.system[color].value};

  padding: ${({ theme, hasTitle, hasActions, hasIcon, size }) => {
    if (hasTitle || hasActions) {
      return theme.sizes.xs.value;
    }
    if (!hasTitle && !hasActions && !hasIcon && size === AlertInlineSize.Compact) {
      return `${theme.sizes['4xs'].value} ${theme.sizes['2xs'].value}`;
    }

    return theme.sizes['2xs'].value;
  }};
`;

export const StyledTextContainer = styled.div``;

export const StyledActionContainer = styled.div`
  display: flex;
  column-gap: ${({ theme }) => theme.sizes['2xs'].value};
  justify-content: flex-end;
  box-sizing: border-box;
  height: 100%;
  align-items: center;
  margin-top: ${({ theme }) => theme.sizes['2xs'].value};
`;

export const StyledActionItem = styled.div`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  box-sizing: border-box;
  button {
    text-decoration: 'underline';
  }
`;

export const StyledContent = styled.div`
  display: flex;
  align-items: center;
  column-gap: ${({ theme }) => theme.sizes['2xs'].value};
`;
export const StyledTitle = styled.div`
  margin-bottom: ${({ theme }) => theme.sizes['2xs'].value};
`;
