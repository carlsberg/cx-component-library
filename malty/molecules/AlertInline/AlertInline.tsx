import { Icon, IconColor, IconSize } from '@carlsberggroup/malty.atoms.icon';
import { Link, LinkColor, LinkStyle } from '@carlsberggroup/malty.atoms.link';
import { Text, TextColor, TextStyle } from '@carlsberggroup/malty.atoms.text';
import { globalTheme as defaultTheme } from '@carlsberggroup/malty.theme.malty-theme-provider';
import React, { useContext } from 'react';
import { ThemeContext } from 'styled-components';
import {
  StyledActionContainer,
  StyledActionItem,
  StyledAlertInLine,
  StyledAlertInLineWrapper,
  StyledContent,
  StyledTextContainer,
  StyledTitle
} from './AlertInline.styled';
import { AlertInlineColor, AlertInlineProps, AlertInlineSize } from './AlertInline.types';

export const AlertInline = ({
  icon,
  size = AlertInlineSize.Default,
  color = AlertInlineColor.Notification,
  dataQaId,
  firstAction,
  firstActionText,
  secondAction,
  secondActionText,
  message,
  title,
  ...props
}: AlertInlineProps) => {
  const theme = useContext(ThemeContext) || defaultTheme;
  const labelFontColor =
    color === AlertInlineColor.Notification || color === AlertInlineColor.Fail
      ? TextColor.White
      : TextColor.DigitalBlack;
  const alertIconColor =
    color === AlertInlineColor.Notification || color === AlertInlineColor.Fail
      ? IconColor.White
      : IconColor.DigitalBlack;
  const actionButtonColor = color === AlertInlineColor.Notification || color === AlertInlineColor.Fail;

  const onFirstAction = () => {
    if (firstAction) {
      firstAction();
    }
  };

  const onSecondAction = () => {
    if (secondAction) {
      secondAction();
    }
  };

  const renderIcon = () => (
    <Icon
      className="inline-alert-icon"
      name={icon}
      size={IconSize.Small}
      color={alertIconColor}
      data-testid={`${dataQaId}-icon`}
    />
  );

  const renderLabel = () => (
    <StyledTextContainer data-alert-label-container>
      <Text
        color={labelFontColor}
        textStyle={size === AlertInlineSize.Default ? TextStyle.MediumSmallDefault : TextStyle.SmallDefault}
      >
        {message}
      </Text>
    </StyledTextContainer>
  );

  const renderActions = () => (
    <StyledActionContainer theme={theme} data-testid={`${dataQaId}-action-container`}>
      <StyledActionItem theme={theme}>
        <Link
          color={actionButtonColor ? LinkColor.White : LinkColor.DigitalBlack}
          linkStyle={LinkStyle.MediumSmallBold}
          onClick={onFirstAction}
          dataTestId={`${dataQaId}-first-action`}
        >
          {firstActionText}
        </Link>
      </StyledActionItem>
      <StyledActionItem theme={theme}>
        <Link
          color={actionButtonColor ? LinkColor.White : LinkColor.DigitalBlack}
          linkStyle={LinkStyle.MediumSmallBold}
          onClick={onSecondAction}
          dataTestId={`${dataQaId}-second-action`}
        >
          {secondActionText}
        </Link>
      </StyledActionItem>
    </StyledActionContainer>
  );

  return (
    <StyledAlertInLineWrapper theme={theme} {...props}>
      <StyledAlertInLine
        hasTitle={!!title}
        hasActions={!!firstActionText || !!secondActionText}
        hasIcon={!!icon}
        size={size}
        data-testid={`${dataQaId}-alert-container`}
        color={color}
        theme={theme}
      >
        {title && (
          <StyledTitle theme={theme}>
            <Text textStyle={TextStyle.MediumSmallBold} color={labelFontColor}>
              {title}
            </Text>
          </StyledTitle>
        )}
        <StyledContent theme={theme} data-testid={`${dataQaId}-alert-content`}>
          {!title && !firstActionText && !secondActionText && icon && renderIcon()}
          {renderLabel()}
        </StyledContent>

        {(firstActionText || secondActionText) && renderActions()}
      </StyledAlertInLine>
    </StyledAlertInLineWrapper>
  );
};
