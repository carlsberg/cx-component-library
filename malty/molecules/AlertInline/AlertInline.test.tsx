import { render } from '@carlsberggroup/malty.utils.test';
import { fireEvent, screen } from '@testing-library/react';
import React from 'react';
import { AlertInline } from './AlertInline';
import { AlertInlineColor } from './AlertInline.types';

describe('alertInline', () => {
  it('should render In Line Alert component with label', () => {
    render(
      <AlertInline
        message="Hello, Im the In Line Alert! Play with me."
        color={AlertInlineColor.Notification}
        dataQaId="inline"
      />
    );
    expect(screen.getByText('Hello, Im the In Line Alert! Play with me.')).toBeInTheDocument();
    expect(screen.getByTestId('inline-alert-container')).toBeInTheDocument();
  });

  it('should render In Line Alert component with label and two actions', () => {
    const secondActionMock = jest.fn();
    render(
      <AlertInline
        message="Hello, Im the In Line Alert! Play with me."
        color={AlertInlineColor.Notification}
        dataQaId="inline-alert-with-icon-and-actions"
        firstActionText="ok"
        secondActionText="cancel"
        secondAction={secondActionMock}
      />
    );

    expect(screen.getByText('ok')).toBeInTheDocument();
    expect(screen.getByText('cancel')).toBeInTheDocument();
    expect(screen.getByText('Hello, Im the In Line Alert! Play with me.')).toBeInTheDocument();

    fireEvent.click(screen.getByText('cancel'));
    expect(secondActionMock).toHaveBeenCalledTimes(1);
  });
});
