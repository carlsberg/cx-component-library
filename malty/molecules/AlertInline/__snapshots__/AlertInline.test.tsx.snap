// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alertInline matches snapshot 1`] = `
<div
  className="sc-kDDrLX sc-iqcoie klQtIm fmCTLE"
>
  <div
    className="sc-crXcEl sc-evZas lcuFxT cieHDJ"
    color="notification-strong"
    data-testid="inline-alert-container"
    size="medium"
  >
    <div
      className="sc-fnykZs iQJZjt"
      data-testid="inline-alert-content"
    >
      <div
        className="sc-breuTD"
        data-alert-label-container={true}
      >
        <span
          className="sc-bczRLJ hROJNc"
        >
          <p
            className="sc-jqUVSM kCAFTy"
            color="white"
          >
            Hello, Im the In Line Alert! Play with me.
          </p>
        </span>
      </div>
    </div>
  </div>
</div>
`;
